sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -446,3 +463,140 @@ def test_string_target_early_stopping(scoring):
     y = np.array(['x'] * 50 + ['y'] * 50, dtype=object)
     gbrt = HistGradientBoostingClassifier(n_iter_no_change=10, scoring=scoring)
     gbrt.fit(X, y)
+
+
+def test_non_uniform_weights_toy_edge_case_reg():
```

Comment:
Not sure what non uniform means.

`test_zero_sample_weights_regression`?

And then small comment at the top like

> Make sure setting a SW to zero amounts to ignoring the corresponding sample